Gems/ROS2/Code/Source/Gripper/VacuumGripperComponent.cpp

```diff
@@ -0,0 +1,352 @@
+                if (foundBody.first != AzPhysics::InvalidSceneHandle)
+                {
+                    AzPhysics::SimulatedBodyEvents::RegisterOnTriggerEnterHandler(
+                        foundBody.first, foundBody.second, m_onTriggerEnterHandler);
+                    AzPhysics::SimulatedBodyEvents::RegisterOnTriggerExitHandler(foundBody.first, foundBody.second, m_onTriggerExitHandler);
+                }
+            }
+
+            AZ::EntityId rootArticulationEntity = GetRootOfArticulation(m_gripperEffectorArticulationLink);
+            AZ::Entity* rootEntity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(rootEntity, &AZ::ComponentApplicationRequests::FindEntity, rootArticulationEntity);
+
+            AZ_Trace("VacuumGripper", "Root articulation entity name: %s\n", rootEntity->GetName().c_str());
+
+            PhysX::ArticulationLinkComponent* component = rootEntity->FindComponent<PhysX::ArticulationLinkComponent>();
+            AZStd::vector<AzPhysics::SimulatedBodyHandle> articulationHandles = component->GetSimulatedBodyHandles();
+
+            AZ_Assert(articulationHandles.size() > 1, "Expected more than one body handles in articulations");
+            for (AzPhysics::SimulatedBodyHandle handle : articulationHandles)
+            {
+                AzPhysics::SimulatedBody* body = sceneInterface->GetSimulatedBodyFromHandle(defaultSceneHandle, handle);
+                AZ_Assert(body, "Expected valid body pointer");
+                if (body->GetEntityId() == m_gripperEffectorArticulationLink)
+                {
+                    m_gripperEffectorBodyHandle = handle;
+                }
+            }
+        }
+        if (m_tryingToGrip)
+        {
+            TryToGripObject();
+        }
+    }
+    AZ::EntityId VacuumGripperComponent::GetRootOfArticulation(AZ::EntityId entityId)
+    {
+        AZ::EntityId parentEntityId{ AZ::EntityId::InvalidEntityId };
+        AZ::Entity* parentEntity = nullptr;
+        AZ::TransformBus::EventResult(parentEntityId, entityId, &AZ::TransformBus::Events::GetParentId);
+        AZ::ComponentApplicationBus::BroadcastResult(parentEntity, &AZ::ComponentApplicationRequests::FindEntity, parentEntityId);
+
+        if (parentEntity == nullptr)
+        {
+            return AZ::EntityId(AZ::EntityId::InvalidEntityId);
+        }
+
+        // Get articulation link component, if not found for parent, return current entity
+        PhysX::ArticulationLinkComponent* component = parentEntity->FindComponent<PhysX::ArticulationLinkComponent>();
+        if (component == nullptr)
+        {
+            return entityId;
+        }
+        return GetRootOfArticulation(parentEntity->GetId());
+    }
+
+    bool VacuumGripperComponent::isObjectGrippable(const AZ::EntityId entityId)
+    {
+        bool isGrippable = false;
+        LmbrCentral::TagComponentRequestBus::EventResult(isGrippable, entityId, &LmbrCentral::TagComponentRequests::HasTag, GrippableTag);
```

Comment:
I am not sure if this solution is the best, since it requires tagging objects for the gripper to work. Is there a better alternative? This is outside of the scope of PR.